src/middleware/withCache.ts

```
@@ -0,0 +1,22 @@
+ * Middleware to add caching headers to API responses
+ *
+ * @param handler The route handler to wrap
+ * @returns A wrapped handler that adds caching headers
+ */
+export function withCache(handler: Function) {
+  return async (request: NextRequest, context: any) => {
+    // Call the original handler
+    const response = await handler(request, context);
+
+    // Add caching headers
+    if (response && response.headers && typeof response.headers.set === 'function') {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)